### PR TITLE
Teach ScrollableList about scroll anchors

### PR DESCRIPTION
**Pull request by abarth**

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie Still needs tests

File: packages/flutter/test/widget/scrollable_list_horizontal_test.dart [hunk 1]
```diff
@@ -27,9 +28,10 @@ Widget buildFrame() {
 }
 
 void main() {
-  test('Drag horizontally', () {
+  test('Drag horizontally with scroll anchor at top', () {
     testWidgets((WidgetTester tester) {
-      tester.pumpWidget(buildFrame());
+      tester.pumpWidget(new Container());
```
Comment (Hixie) [inline packages/flutter/test/widget/scrollable_list_horizontal_test.dart]: you shouldn't need this. pumpWidget() pumps an ErrorWidget between each pump just to clear the air.

Comment (Hixie) [inline packages/flutter/test/widget/scrollable_list_horizontal_test.dart]: or rather, testWidgets does.

File: packages/flutter/test/widget/scrollable_list_horizontal_test.dart [hunk 2]
```diff
@@ -147,4 +149,107 @@ void main() {
       expect(tester.findText('5'), isNull);
     });
   });
+
+  test('Drag horizontally with scroll anchor at end', () {
+    testWidgets((WidgetTester tester) {
+      tester.pumpWidget(new Container());
```
Comment (Hixie) [inline packages/flutter/test/widget/scrollable_list_horizontal_test.dart]: ditto

File: packages/flutter/test/widget/scrollable_list_horizontal_test.dart [hunk 3]
```diff
@@ -147,4 +149,107 @@ void main() {
...
+      //   520..810 = 4
+      expect(tester.findText('0'), isNull);
+      expect(tester.findText('1'), isNull);
+      expect(tester.findText('2'), isNotNull);
+      expect(tester.findText('3'), isNotNull);
+      expect(tester.findText('4'), isNotNull);
+      expect(tester.findText('5'), isNull);
+
+      // the center of item 3 is visible, so this works;
+      // if item 3 was a bit wider, such that it's center was past the 800px mark, this would fail,
```
Comment (Hixie) [inline packages/flutter/test/widget/scrollable_list_horizontal_test.dart]: its

Comment (Hixie) [inline packages/flutter/test/widget/scrollable_list_horizontal_test.dart]: fix my typo in the earlier comment too...

**Comment by Hixie** [comment]
lgtm

**Pull request merged**